python/paddle/fluid/tests/unittests/dygraph_to_static/test_print.py

```diff
@@ -0,0 +1,233 @@
+                    args=[Name(id='x_v', annotation=None, type_comment=None)],
+                    keywords=[])],
+            keywords=[]))
+    """
+    x_v = fluid.dygraph.to_variable(x)
+    print("PrintVariable: {}".format(x_v))
+
+
+# 4. print VarBase with format 2
+@declarative
+def dyfunc_print_with_format2(x):
+    """
+    PY2:
+    Print(dest=None,
+        values=[
+            BinOp(left=Constant(value='PrintVariable: %s', kind=None),
+                op=Mod,
+                right=Name(id='x_v', annotation=None, type_comment=None))],
+        nl=True)
+    PY3:
+    Expr(
+        value=Call(func=Name(id='print', annotation=None, type_comment=None),
+            args=[
+                BinOp(left=Constant(value='PrintVariable: %s', kind=None),
+                    op=Mod,
+                    right=Name(id='x_v', annotation=None, type_comment=None))],
+            keywords=[]))
+    """
+    x_v = fluid.dygraph.to_variable(x)
+    print("PrintVariable: %s" % (x_v))
+
+
+# 5. print VarBase in control flow1
+@declarative
+def dyfunc_print_with_ifelse(x):
+    x_v = fluid.dygraph.to_variable(x)
+    if len(x_v.shape) > 1:
+        print(x_v)
+    else:
+        print(x_v)
+
+
+# 6. print mutiple VarBases
+@declarative
+def dyfunc_print_multi_vars(x):
+    """
+    # NOTE: y_v type is error before cur PR in this case
+    Assign(targets=[Name(id='y_v', annotation=None, type_comment=None)],
+        value=BinOp(left=Name(id='x_v', annotation=None, type_comment=None), op=Mult, right=Constant(value=2, kind=None)))
+    """
+    x_v = fluid.dygraph.to_variable(x)
+    y_v = x_v * 2
+    print(x_v)
+    print(y_v)
+
+
+# 7. print continue VarBase
+@declarative
+def dyfunc_print_continue_vars(x):
+    """
+    PY3:
+    Expr(
+        value=Call(func=Name(id='print', annotation=None, type_comment=None),
+            args=[Name(id='x_v', annotation=None, type_comment=None),
+                Name(id='y_v', annotation=None, type_comment=None)],
+            keywords=[]))
+    PY2:
+    Print(dest=None,
+        values=[
+            Tuple(
+                elts=[Name(id='x_v', annotation=None, type_comment=None),
+                    Name(id='y_v', annotation=None, type_comment=None)])],
+        nl=True)
+    """
+    x_v = fluid.dygraph.to_variable(x)
+    y_v = x_v * 2
+    print(x_v, y_v)
+
+
+class TestPrintBase(unittest.TestCase):
+    def setUp(self):
+        self.input = numpy.ones(5).astype("int32")
+        self.place = fluid.CUDAPlace(0) if fluid.is_compiled_with_cuda(
+        ) else fluid.CPUPlace()
+        self.set_test_func()
+
+    def set_test_func(self):
+        raise NotImplementedError("Print test should implement set_test_func")
+
+    def get_dygraph_output(self):
+        with fluid.dygraph.guard():
+            self.dygraph_func(self.input)
+
+    def get_static_output(self):
+        with fluid.program_guard(fluid.Program()):
+            # TODO: How to catch C++ stdout to python
```

Comment:
You could try the first answer at the following link:

https://stackoverflow.com/questions/33767627/python-write-unittest-for-console-print

Reply:
I tried this, it only can deal capture python `print`, but cannot capture `std::cout` in `layers.Print`

Reply:
ok ... than it's OK for me to put it as TODO